src/compiler/utils.c

```
@@ -0,0 +1,93 @@
+}
+
+/**
+ * Writes an 32 bit number into the file.
+ */
+void write32(FILE *file, uint32_t value) {
+        write8(file, (value >> 0)  & 0xff);
+        write8(file, (value >> 8)  & 0xff);
+        write8(file, (value >> 16) & 0xff);
+        write8(file, (value >> 24) & 0xff);
+
+	printf("32 Input for %d: %02x %02x %02x %02x\n", (value >> 0) & 0xff, (value >> 8) & 0xff, (value >> 16) && 0xff, (value >> 24) & 0xff);
```

cppcheck (2.10-2)

[error] printf format string requires 5 parameters but only 4 are given.

(wrongPrintfScanfArgNum)